src/include/daos/tests_lib.h

```diff
@@ -228,18 +228,19 @@ int dmg_pool_create(const char *dmg_config_file,
 		    d_rank_list_t *svc, uuid_t uuid);
 
 /**
- * Destroy a pool with \a uuid. If there is at least one connection to this
- * pool, and \a force is zero, then this operation completes with DER_BUSY.
- * Otherwise, the pool is destroyed when the operation completes.
+ * Destroy a pool with \a uuid. If there is at least one connection to this pool, and \a force is
+ * zero, then this operation completes with DER_BUSY. If there is at least one container in this
+ * pool, and \a recursive is zero, then this operation completes with DER_MISC. Otherwise, the
```

Comment:
Hmm. `-DER_MISC` doesn't seem right for this. Maybe other reviewers will have a better idea, but seems like `-DER_EXIST` or `-DER_BUSY` might give the user more of a hint as to what the problem is?

Reply:
So this refers to the interpretation of the dmg pool destroy output by a C test tool. How should the C code interpret the faultt output from dmg to determine DER_BUSY? should it read the dmg stdout to check for a given text string? that sounds less than ideal. I think leaving it as DER_MISC is the best option here.. what do you think?

Reply:
Does the C test tool do any interpretation of the result? Pretty sure it just fails. In any case, I don't think we should make decisions about return codes based on what's easiest for the tests. We should anticipate that this result may eventually be seen by a human who will likely be more confused by `-DER_MISC` than `-DER_BUSY`. Even clearer would be a new errno, e.g. `-DER_NOTEMPTY`, but I guess we're trying to avoid adding to that set.

Reply:
In the daos_test case, I don't think there's a reason to run dmg pool destroy without --recursive. We always want to clean up, so passing false would be a mistake. I don't think we should allow it at all here.

Reply:
@kjacque originally I thought the same but then I imagined someone would probably ask me to add the flexibility in the helper method to explicitly set the parameter, hence adding the option. 

@mjmac the test tool extracts the status from the JSON response, the user will see the dmg fault if containers exist on the pool but the recursive flag is not set.

I will remove the option to set recursive value in the dmg_pool_destroy helper, hopefully that's acceptable and will enable this PR to be landed?

Reply:
Yeah, as discussed, I don't think we need to have that level of flexibility in the C tests. There should be some ftest coverage to verify the container check, though.